### PR TITLE
send buildpack_provided_description (language-pack) with new slugs

### PR DESCRIPTION
**Pull request by hunterloftis**
resolves https://github.com/heroku/heroku-docker/issues/118

### CODE REVIEW

**Comment by freeformz** [comment]
I'm -1 on using app.json for this and getting the various FROMS from docker or extracting the FROM line of the Dockerfile that is actually building the image.

With that said, this is better than the current situation, so I'm fine with it as is if we don't want to go that far.

**Comment by hunterloftis** [comment]
>  getting the various FROMS from docker

Is there a straightforward way to do this?

**Comment by freeformz** [comment]
We may also want to consider this:

heroku-docker (<IMAGE>  |  <FROMS>)
heroku-docker (unknown)  # in the unknown case

**Comment by friism** [comment]
@hunterloftis I just spent some poking around at this, I can't find a good way to query this. The most accurate would probably be grabbing something out of `docker history {image}`.

**Comment by freeformz** [comment]
we can get metadata from the docker images via docker inspect. We can add metadata to our images with the LABEL flag inside of our Dockerfiles

**Comment by hunterloftis** [comment]
As a buildpack maintainer, one of the dangers of stuff like the LABEL flag is that there's no authoritative way to know if a self-reported LABEL is true. For instance, there's no easy way for me to differentiate between the official node buildpack and a forked 3rd party buildpack because they self-report their own names.

That's why the Docker image is appealing as a canonical descriptor.

**Comment by freeformz** [comment]
Yes, but they're both node, or go, or perl or whatever.

**Comment by hunterloftis** [comment]
Theoretically ;)

File: commands/release.js [hunk 1]
```diff
@@ -133,7 +134,8 @@ function release(context) {
     cli.log('creating remote slug...');
     cli.log('remote process types:', modifiedProc);
     var slugInfo = app.slugs().create({
-      process_types: modifiedProc
+      process_types: modifiedProc,
+      buildpack_provided_description: appJSON.image || 'docker (unknown)'
```
Comment (freeformz) [inline commands/release.js]: At the very least I'd like to consider changing this to:

"heroku-docker (appJSON.image || unknown)"

**Comment by hunterloftis** [comment]
It isn't super clean but I could probably parse the output here:

```
Building web...
Step 0 : FROM heroku/go:1.5
# Executing 2 build triggers
Trigger 0, COPY . /app/.temp
Step 0 : COPY . /app/.temp
 ---> Using cache
```

looking for "FROM x"

**Comment by freeformz** [comment]
FWIW: I'm fine with using app.json's "image" for now and perhaps revisiting later. Trying to not let "perfect" be the obstacle to "better".

Using the output of the build would be a good though IMO.

**Comment by friism** [comment]
Since the Docker builder can get a hold of that, I'm sure we can too.

What about iterating back across `docker history` and `docker inspect {}` until we find the first FROM.

**Comment by freeformz** [comment]
hoot!

**Pull request merged**